Remove FutureMessage from sender ProcessGroupAgent

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #50236 Apply clang-format to rpc cpp files
* #50029 Completely remove FutureMessage type
* #50028 Completely Remove FutureMessage from RPC agents
* #50027 Completely Remove FutureMessage from RPC cpp tests
* #50026 Remove FutureMessage from RPC request callback logic
* #50025 Completely remove FutureMessage from FaultyProcessGroupAgent
* #50024 Remove FutureMessage from sender TensorPipeAgent
* **#50023 Remove FutureMessage from sender ProcessGroupAgent**
* #50020 Completely remove FutureMessage from distributed autograd
* #50005 Completely remove FutureMessage from RPC TorchScript implementations
* #50004 Completely remove FutureMessage from RRef Implementations
* #49995 Replace FutureMessage with ivalue::Future in RpcAgent retry logic
* #49960 Replace FutureMessage with ivalue::Future in RRefContext
* #49927 Replace FutureMessage with ivalue::Future in distributed/autograd/utils.*
* #49906 Let RpcAgent::send() return JitFuture

Differential Revision: [D25753217](https://our.internmc.facebook.com/intern/diff/D25753217)

mrshenli committed:
Remove FutureMessage from sender ProcessGroupAgent

[ghstack-poisoned]

mrshenli committed:
Update on "Remove FutureMessage from sender ProcessGroupAgent"


Differential Revision: [D25753217](https://our.internmc.facebook.com/intern/diff/D25753217)

[ghstack-poisoned]

mrshenli committed:
Update on "Remove FutureMessage from sender ProcessGroupAgent"


Differential Revision: [D25753217](https://our.internmc.facebook.com/intern/diff/D25753217)

[ghstack-poisoned]

lw commented on torch/csrc/distributed/rpc/process_group_agent.cpp:
```diff
@@ -591,10 +590,14 @@ bool ProcessGroupAgent::handleRecv(RecvWork& work) {
     if (message.type() == MessageType::EXCEPTION) {
-      fm->setError(
-          std::string(message.payload().begin(), message.payload().end()));
+      //fm->setError(
+      //    std::string(message.payload().begin(), message.payload().end()));
```
Any reason the old code is kept around as a comment?

mrshenli replied:
ahh, sorry, that's my debug vestige. removing

mrshenli committed:
Update on "Remove FutureMessage from sender ProcessGroupAgent"


Differential Revision: [D25753217](https://our.internmc.facebook.com/intern/diff/D25753217)

[ghstack-poisoned]

mrshenli committed:
Update on "Remove FutureMessage from sender ProcessGroupAgent"


Differential Revision: [D25753217](https://our.internmc.facebook.com/intern/diff/D25753217)

[ghstack-poisoned]
